File: gRPC/models/Course.js
const mongoose = require("mongoose");

const CourseSchema = new mongoose.Schema(
  {
    id: {
      type: String,
      required: [true, "Please add an ID"],
    },
    CourseName: {
      type: String,
      required: [true, "Please add a name"],
    },
    CourseCode: {
      type: String,
      required: [true, "Please add a price"],
    },
  },
  {
    toJSON: { virtuals: true },
    toObject: { virtuals: true },
  }
);

module.exports = mongoose.model("Course", COurseSchema);

File: managementservice/Course.go
package models

import "gorm.io/gorm"

type Course struct {
	gorm.Model
	CourseName string `gorm:"type:varchar(255);notNull" json:"course_name"`
	CourseCode string `gorm:"type:varchar(255);uniqueIndex;notNull" json:"course_code"`
}
